grants/speculative/zeroknowledgeidentity.md

```diff
@@ -0,0 +1,115 @@
+
+Linking web2 identity
+
+    A user who wishes to link a web2 identity will do the following:
+    1. Register their identity using Edgeware's Identity Service (edgeware-watcher).
+    2. Attest to their identity and include a leaf node Hash(Hash(secret||nullifier)||accountId||web3Identity)
+    3. If the identity passes verification, this leaf node is appended to the SMT
+    4. To prove membership, a user submits a proof proving they know the preimage (secret||nullifier) of the leaf node
+
+***Nullifiers***: nullifiers are needed to prevent re-submitting proofs for auxiliary accounts.
+
+## **Team Members**
+
+- Drew Stone - Lead
+- Dillon Chen
+- Raymond Zhong
+- Thom Ivy
+- Jake Naviasky
+
+## **Team Website**
+
+[https://commonwealth.im/](https://commonwealth.im/) and [https://edgewa.re](https://edgewa.re/)
+
+## **Legal Structure**
+
+incorporated in USA as a Delaware C-Corp
+
+## **Team's experience**
+
+Commonwealth Labs is a blockchain company which is focused on building Edgeware, a WASM-based smart contract parachain as well as supporting governance tools. For Edgeware and Polkadot ecosystem governance, among other things, we've built an identity module, a UI to simply the governance process across parachains. Drew Stone, the lead engineer, has been working as an Ethereum smart contract developer and blockchain developer for 2+ years.
+
+Relevant projects to the goal are below:
+
+- Code
+    - Miximus-style ZK membership proofs: [https://github.com/drewstone/rust-miximus](https://github.com/drewstone/rust-miximus)
+    - Sparse Merkle Tree substrate module: [https://github.com/hicommonwealth/edgeware-node/tree/drew.merkle/modules/edge-merkletree](https://github.com/hicommonwealth/edgeware-node/tree/drew.merkle/modules/edge-merkletree)
+- Articles
+    - [https://medium.com/commonwealth-labs/anonymous-voting-a-design-survey-12de869dc97f](https://medium.com/commonwealth-labs/anonymous-voting-a-design-survey-12de869dc97f)
+
+## **Team Code Repos**
+
+- Edgeware: [http://github.com/hicommonwealth/edgeware-node](http://github.com/hicommonwealth/edgeware-node)
+- Edgeware-watcher: [https://github.com/hicommonwealth/edgeware-watcher](https://github.com/hicommonwealth/edgeware-watcher)
+- Edgeware-identity: [https://github.com/hicommonwealth/edge-identity](https://github.com/hicommonwealth/edge-identity)
+- Rust-mixmus: [https://github.com/drewstone/rust-miximus](https://github.com/drewstone/rust-miximus)
+
+## **Team LinkedIn Profiles**
+
+[https://www.linkedin.com/in/dillchen/](https://www.linkedin.com/in/dillchen/)
+
+[https://www.linkedin.com/in/drewcstone/](https://www.linkedin.com/in/drewcstone/)
+
+[https://www.linkedin.com/in/raykyri/](https://www.linkedin.com/in/raykyri/)
+
+## **Development Roadmap**
+
+The milestones are spread out over a total of 3 months as following:
+
+- M0: Aggregate all the cryptography tools importable in a portable rust library (1 week)
+    - Pairing/bellman/sapling-crypto as pure rust libraries
```

Comment:
Is there some reason every little zk project vendor all their zcash dependencies?  It's bad practice normally

Reply:
There are so few libraries that I chose ones I have experience working with. We can research alternatives though, but these are what I am familiar with in the Rust ecosystem.

Reply:
I meant why do you and other projects pull them into your own repo?  what are you changing in them?